lib/ansible/module_utils/ansible_tower.py

```diff
@@ -42,33 +42,37 @@ def tower_auth_config(module):
     '''tower_auth_config attempts to load the tower-cli.cfg file
     specified from the `tower_config_file` parameter. If found,
     if returns the contents of the file as a dictionary, else
-    it will attempt to fetch values from the module pararms and
-    only pass those values that have been set.
+    it will attempt to pop the values from the module parameters.
+    It will only pass the values that might have been set and will clean
+    the dictionary from the tower keys created in tower_argument_spec,
+    so that when this function completes, the module.params dictionary will
+    not contain any keys related to the tower authentication.
     '''
+    tower_auth_args = [
+        'tower_host',
+        'tower_username',
+        'tower_password',
+        'tower_verify_ssl',
+        'tower_config_file',
```

Comment:
`tower_config_file` doesn't need to be in `tower_auth_args` (this value is always removed from `module.params`)